src/Resource_Element.hxx

```diff
@@ -311,14 +326,56 @@ public:
         get_main_table_element().set_data(d);
     }
 
+    Resource_Type get_resource_type() const { return resource_type_; };
+
+    bool is_results_resource() const {
+        return (resource_type_ == Resource_Type::RESULTS);
+    }
+
+
 private:
+    static inline Resource_Type get_type_enum(const std::string &type_str) {
+        if (boost::equals(type_str, "results")) {
+            return Resource_Type::RESULTS;
+        }
+        if (boost::equals(type_str, "meta")) {
+            return Resource_Type::META;
+        }
+        return Resource_Type::OTHER;
+    }
+
+
+    tablator::Resource_Type determine_resource_type(tablator::ATTRIBUTES attributes) {
+        for (const auto &attr_pair : attributes) {
+            if (boost::iequals(attr_pair.first, tablator::TYPE)) {
+                return tablator::Resource_Element::get_type_enum(attr_pair.second);
+            }
+        }
+        return determine_resource_type_without_attrs();
+    }
+
+    tablator::Resource_Type determine_resource_type_without_attrs() {
+        if (!table_elements_.empty() &&
+            !get_main_table_element().get_columns().empty()) {
+            return tablator::Resource_Type::RESULTS;
+        }
+        return tablator::Resource_Type::OTHER;
+    }
```

Comment:
This worries me a little, since the 2 resource type determinations can disagree, but I guess it's needed to deal with tables that are not following the `type="results"` convention?

Reply:
Well, I only call the `without_attrs()` version if there are no attributes, and`with`  calls `without`.  I toyed with rewriting the `without_attrs()` version to create an empty ATTRIBUTEs and return
`determine_resource_type(empty_attrs)`; would that be clearer?

Reply:
That might be a bit clearer, but does involve creation of an unnecessary object. Actually now that I am looking at it again, shouldn't the without case take a const-reference to a `tablator::ATTRIBUTES` to avoid a copy?

Reply:
The `with` case, you mean.  Yup!